provider/google.go

```diff
@@ -123,6 +118,7 @@ func NewGoogleProvider(project string, dryRun bool) (Provider, error) {
 
 	provider := &googleProvider{
 		project: project,
+		domain:  domain,
```

Comment:
I noticed that there's quite some duplication with respect to detecting zones and sanitizing records and I have that on my TODO list. However, I think none of this would change anything from the user's perspective (flags and created DNS records), right?! If that's the case I would leave that for our post v0.3 stabilization phase.